File: js/SingleNumber2.js
var singleNumber = function(nums) {
    let digit_set = new Set(nums);
    let digit_sum = 0;
    for(let n of digit_set){
        digit_sum += n;
    }
    let num_sum = nums.reduce(function(sum, val){ return sum + val;});
    return (digit_sum*3 - num_sum)/2;
};

singleNumber([2, 2, 3, 2]);

File: csharp/SingleNumber2.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace LeetCode
{
    class SingleNumber2
    {
        public int SingleNumber(int[] nums)
        {
            checked
            {
                HashSet<int> digitSet = new HashSet<int>(nums);
                long digitSum = 0;
                foreach (var i in digitSet)
                {
                    digitSum += i;
                }
                long numsSum = 0;
                foreach (var i in nums)
                {
                    numsSum += i;
                }
                long result = (digitSum * 3 - numsSum) / 2;
                return (int)result;
            }
        }
    }
}
